.github/workflows/feg-workflow.yml

```diff
@@ -0,0 +1,70 @@
+name: feg-workflow
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+########## FEG-PRECOMMIT ##########
+
+  feg-precommit:
+
+    runs-on: ubuntu-latest
+
+    env:
+      GO111MODULE: on
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Export Magma root path
+      run: echo "MAGMA_ROOT=${PWD}" >> $GITHUB_ENV
+    - name: Run golang_before_install.sh script
+      run: ./circleci/golang_before_install.sh
+    - name: Run go mod download with retry
+      uses: nick-invision/retry@v2
+      with:
+        command: cd ${MAGMA_ROOT}/cwf/gateway && go mod download
+        timeout_minutes: 10
+    - name: make feg precommit
+      run: |
+          cd ${MAGMA_ROOT}/feg/gateway
+          make -C ${MAGMA_ROOT}/feg/gateway precommit
+#    - magma_slack_notify
+
+########## FEG-BUILD ##########
+
+  feg-build:
+
+    needs: feg-precommit
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+#      - build/determinator:
+#          paths: "cwf/k8s"
```

Comment:
Github Actions support limiting workflows to changes to [specific files](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) on pushes and pull requests (though not tags).

I'm not certain if both these jobs are supposed to be triggered when changes happen to files under `cwf/k8s`, or if only _this_ job should be triggered. If it's the latter, then you may need to look into using a [label action](https://github.com/marketplace/actions/labeler) to signify a PR made changes to a file, followed by an action that's triggered on those [labels](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#label).

Reply:
Thanks Trevor!.. I just double checked and all feg jobs should be triggered for paths under *federated_build_verify

In that case, should I still look into the label action?

Reply:
If it's all the jobs, then you should be able to trigger on the [paths](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) and not worry about labeling.